presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -62,14 +64,17 @@
     // for each source, the list of inputs corresponding to each output
     private final List<List<Symbol>> inputs;
 
+    private final Optional<OrderingScheme> orderingScheme;
```

Comment:
Don't you think that merging deserve separate `ExchangeNode.Type=MERGE`? This way we can clearly separate simple gathering from more complex merging. In the future we could use this type to do intermittent merging. What do you think? I don't have strong opinion about this.